### PR TITLE
Feature/NSA 8098 fix orphaned service requests after user org removal

### PR DESCRIPTION
**Pull request by AMcCartneyMTH**
NSA-8098: Fix service/sub-service requests still existing when a user is removed from an organisation.
(https://dfe-secureaccess.atlassian.net/browse/NSA-8098)

This ticket covers the work required to address the defect where service and sub-service requests were becoming orphaned  when a user was removed from an organisation.

As part of the work undertaken to address the defect, additional unit test coverage has been applied to both the happy and sad execution paths.

### CODE REVIEW

File: src/app/organisations/deleteUserOrganisation.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
-const { deleteUserOrganisation } = require('./data/organisationsStorage');
+const { deleteUserOrganisation, getServiceAndSubServiceReqForOrgs, updateUserServSubServRequest } = require('./data/organisationsStorage');
 
 const deleteOrg = async (req, res) => {
-  const organisationId = req.params.id;
-  const userId = req.params.uid;
+  const correlationId = req.get('x-correlation-id');
+  const { id: orgId, uid: userId } = req.params;
```
Comment (malikimran34) [inline src/app/organisations/deleteUserOrganisation.js:5]: it can go like this const { orgId, userId } = req.params;
id, uid are not being used

Comment (malikimran34) [inline src/app/organisations/deleteUserOrganisation.js:5]: is User id being validated as not null or undefined?

Comment (AMcCartneyMTH) [reply src/app/organisations/deleteUserOrganisation.js:6]: A guard clause has been introduced in the implementation for both the orgId and userId params and the tests have been updated to match.

File: src/app/organisations/deleteUserOrganisation.js [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+        request.actioned_reason = 'Rejected';
+        await updateUserServSubServRequest(request.id, request);
+      })
+    );
+
+    // Delete the user organisation mapping
+    await deleteUserOrganisation(orgId, userId, correlationId);
+
+    res.status(204).send();
+  } catch (error) {
```
[the remark this answers is not in the document]
Comment (AMcCartneyMTH) [reply src/app/organisations/deleteUserOrganisation.js:32]: The error message(s) look like:

"Missing required parameter: orgId" and "Missing required parameter: userId"

These do not exceed the maximum length of the logger, but a good shout :)

**Review by AMcCartneyMTH — COMMENTED** [review]
I have made the suggested changes in regards to guard clauses and answered/clarified review points.

**Pull request merged**